### DISCUSSION TITLE
Diarization Not Working on Custom Audio Files

### DISCUSSION THREAD

**Discussion by sherySSH** [NVIDIA-NeMo/NeMo#3238] [Q&A]
Hey!

I am working with this [notebook](https://github.com/NVIDIA/NeMo/blob/offline_asr_diar_vad/tutorials/speaker_tasks/Speaker_Diarization_Inference.ipynb) for diarizing the audio files. This notebook seems to work perfectly with the given "an4_diarize_test.wav" test file but when I try it on my own audio files then model gives runtime error as shown below. I wonder why it is not processing the audio properly and giving the empty tensor. My audio files are also in .wav format, therefore, there is not any inconsistency as far as audio file format is concerned.

![image](https://user-images.githubusercontent.com/32700434/143193719-e0ef20ee-2711-48df-82a9-3dc6c655c36b.png)

I also went through the source code of [audio_to_label.py](https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/asr/data/audio_to_label.py) and I found that audio_signal tensor has 0 elements. Moreover, the audio_lengths is tensor(2) which means it is only processing two samples of audio which does not make sense. Any help would be much appreciated!

**Comment by jqueguiner** [comment]
audio_lengths is tensor(2) probably means that you have 2 channels in your audio.

**Reply by sherySSH** [reply]
Yeah, that was the issue. I had found the solution but forgot to also share it here. Thanks for sharing!
